src/client/manager.rs

```diff
@@ -2,89 +2,162 @@
+        }
     }
 }
 
-impl Manager {
+impl Client {
     pub fn new() -> Self {
         Self {
-            connections: Arc::new(RwLock::new(HashMap::new())),
-            inodes: Arc::new(Mutex::new(HashMap::new())),
-            inodes_reverse: Arc::new(Mutex::new(HashMap::new())),
+            client: rpc::client::Client::default(),
+            inodes: DashMap::new(),
+            inodes_reverse: DashMap::new(),
+            parent_inodes: DashMap::new(),
         }
     }
 
-    pub fn add_connection(&self, id: i32, connection: Connection) {
-        self.connections
-            .write()
-            .unwrap()
-            .insert(id as i32, connection);
+    pub fn add_connection(&self, server_address: &str) {
+        self.client.add_connection(server_address);
+    }
+
+    pub fn remove_connection(&self, server_address: &str) {
+        self.client.remove_connection(server_address);
     }
 
-    pub fn remove_connection(&self, id: i32) {
-        self.connections.write().unwrap().remove(&(id as i32));
+    pub fn get_connection_address(
+        &self,
+        _path: &str,
+    ) -> Result<String, Box<dyn std::error::Error>> {
+        // mock
+        Ok("127.0.0.1:8085".to_string())
     }
 
-    pub fn get_connection_index(&self, path: &str) -> Option<i32> {
-        let hash = hash(path);
-        let index = hash % self.connections.read().unwrap().len() as u64;
-        Some(index as i32)
+    pub fn temp_init(
+        &'static self,
+        server_address: &str,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        debug!("temp init");
+        debug!("add connection");
+        self.add_connection(server_address);
+
+        self.inodes_reverse.insert(1, "/".to_string());
+        self.inodes.insert(
+            "/".to_string(),
+            Node::new(
+                "/".to_string(),
+                "/".to_string(),
+                FileAttr {
+                    ino: 1,
+                    size: 0,
+                    blocks: 0,
+                    atime: UNIX_EPOCH,
+                    mtime: UNIX_EPOCH,
+                    ctime: UNIX_EPOCH,
+                    crtime: UNIX_EPOCH,
+                    kind: fuser::FileType::Directory,
+                    perm: 0o755,
+                    nlink: 0,
+                    uid: 0,
+                    gid: 0,
+                    rdev: 0,
+                    flags: 0,
+                    blksize: 0,
+                },
+            ),
+        );
+        self.parent_inodes.insert(1, 1);
+
+        debug!("spawn parse thread");
+        // spawn a thread to handle the connection using self.parse_response
+        std::thread::spawn(|| self.client.parse_response());
+        Ok(())
     }
 
     pub fn lookup_remote(&self, parent: u64, name: &OsStr, reply: ReplyEntry) {
         log::info!("lookup_remote");
-        if self.inodes_reverse.lock().unwrap().contains_key(&parent) {
+        if self.inodes_reverse.contains_key(&parent) {
```

Comment:
I think it is better to use `get` directly to judge whether the `Option` is `None`

Reply:
Done.